installation_policies/pre_spack_install_policies.md

```diff
@@ -0,0 +1,124 @@
+This document lists the installation policies prior to [PR20](https://github.com/xsdk-project/xsdk-community-policies/pull/20), which changed policy M1 from requiring CMake/Autoconf installation to Spack based installation.
+
+----
+
+1. Implement the default behavior described below. Each package can decide whether XSDK mode is the default mode.<sup>4</sup>
+
+    1. `--dis/enable-xsdk-defaults`
+    2. `USE_XSDK_DEFAULTS=[YES,NO]`
+  
+<sup>4</sup> For packages like Trilinos that need to maintain backward compatibility over consecutive releases, 
+`USE_XSDK_DEFAULTS` may be `FALSE` by default.
+
+----
+
+2. Identify location to install package. Multiple "versions" of packages, such as debug and release, can be installed by 
+using different prefix directories.
+
+    1. `--prefix=directory`
+    2. `CMAKE_INSTALL_PREFIX=directory`
+
+----
+
+3. Select compilers<sup>5</sup> and compiler flags.
+
+    1. If the compilers (and or flags) are explicitly set on input, use those:
+
+        1. `CC=<cc>`,  `CXX=<cxx>`,  `FC=<fc>`,  `CPP`, `FFLAGS`, `FCFLAGS`, `CFLAGS`, `CXXFLAGS`, `CPPFLAGS`, `LDFLAGS`
+
+        2. `CMAKE_C_COMPILER=<cc>`, `CMAKE_CXX_COMPILER=<cxx>`, `CMAKE_Fortran_COMPILER=<fc>`, `CMAKE_C_FLAGS="<flag1> <flag2> ...”`, `CMAKE_CXX_FLAGS=”...”`, `CMAKE_Fortran_FLAGS=”...”`
+
+    2. If the compilers and/or flags are not explicitly set on input but are set in the environment variables
+    `FC`, `CC`, `CXX`, `CPP`<sup>6</sup>, `FFLAGS`, `FCFLAGS`, `CFLAGS`, `CXXFLAGS`, `CPPFLAGS`,<sup>7</sup>
+    `LDFLAGS`, then the compilers and flags must be set to these.  If both `FFLAGS` and `FCFLAGS` are set, then they
+    need to be the same or it is an error.
+
+    3. If the compilers and/or compiler flags are not explicitly passed in or defined in the environment variables listed 
+    above, then the package is free to try to find compilers on the system and set the compiler flags consistent with the 
+    other settings defined below (e.g., shared libraries vs. static libraries, debug vs. non-debug).<sup>8</sup>
+
+<sup>5</sup> Packages must support using the MPI compiler wrappers for these arguments.
```

Comment:
This doesn't work with HIP and some CUDA packages: 
* Kokkos for example requires the use of nvcc-wrapper which cannot be set to mpicxx. 
* CMake currently has very bad HIP support, which means that compiling HIP kernels with CMake requires the use of the CMAKE_CXX_COMPILER=hipcc compiler.

Reply:
This document is the installation policies as they currently exist, it is meant only to be an archive.